odoo/_monkeypatches/__init__.py

```diff
@@ -11,29 +14,69 @@ def set_timezone_utc():
         time.tzset()
 
 
-def patch_all():
+class PatchImportHook:
+    """Register hooks that are run on import."""
+
+    def __init__(self):
+        self.hooks = {}
+
+    def add_hook(self, fullname: str, hook):
+        """Register a hook after a module is loaded.
+        If already loaded, run hook immediately."""
+        self.hooks[fullname] = hook
+        if fullname in sys.modules:
+            hook()
+
+    def find_spec(self, fullname, path=None, target=None):
+        if (hook := self.hooks.get(fullname)) is None:
+            return
+
+        # skip all finders before this one
+        found_self = False
+        for finder in sys.meta_path:
+            if not found_self:
+                found_self = finder is self
+                continue
+            spec = finder.find_spec(fullname, path, target)
+            if spec is not None:
+                # we found a spec, change the loader
+                loader = spec.loader
+
+                class HookLoader:
+                    def load_module(self, fullname):
+                        module = loader.load_module(fullname)
+                        hook()
+                        return module
```

Comment:
:nerd_face: 

but frankly this is only because I searched for this `hook` variable 5 seconds too much

```suggestion
                loader = spec.loader

                class HookLoader:
                    def load_module(self, fullname, hook=hook, load_module=spec.loader.load_module)
                        module = load_module(fullname)
                        hook()
                        return module
```

Reply:
I think `hook=hook` won't help understanding the code. I still added the `load_module_parameter`.